src/styles/index.js

```diff
@@ -109,3 +109,27 @@ export const Dot = styled.span`
 export const FallbackSpan = styled.span`
   opacity: 0.5;
 `;
+
+export const UpgradeNoticeStyles = ({
```

Comment:
Is there any reason why this is a mixin and not a component?

Reply:
Wanted to decouple styles and markup/functionality to avoid what has happenend with AZLabels.

Reply:
But it's being used like a component. The separate `text-decoration` values can be implemented by extending the component

Reply:
Yes it can, but then we need to pass a function and we need to add this function to the component and check that is not null, etc.

I prefer this approach because is the simpler by now. And I'm not done BTW.

Reply:
I'm not sure I follow... extending is pretty easy with styled components

@oponder  had a similar situation in this file, where he extended the component (Check the `Copyable` component):
https://github.com/giantswarm/happa/blob/master/src/components/UI/AppDetails/ChartVersionsTable.js

Reply:
I wasn't following you either  😆 🤦‍♂ So I was doing a mixin instead of and extend because my initial idea was to have a bunch of styles to throw anywhere we need them, then I noticed that I had to customize one property and did it this way, just because sass. Any special reason.

Do we have guidelines about this?

Reply:
or opinions?

Reply:
We haven't documented this yet.

I believe we should only use mixins when they can make reusability easier across different contexts, and when they can be composed with other styles to create components. Ex: a `CenteredFlexContainer` mixin would return the styles:
```css
display: flex;
justify-content: center;
align-items: center;
```

Reply:
Can we have this discussion in an a dedicated PR instead? Or in `fmt` repo? So we can mege this

Reply:
Yeah I think there is an easier to follow way that I also recently learned on how to deal with these cases.

If you wanna pair on a refactor Im happy to, then we can also make a fmt PR.

I think we can take the time to 'get it right' in this PR though. Or is this change is urgent?